tests/png/bug00295.c

```diff
@@ -0,0 +1,39 @@
+/**
+ * Regression test for <https://github.com/libgd/libgd/issues/295>.
+ *
+ * We're testing that reading a PNG which might trigger a libpng warning
+ * does not trigger a GD_ERROR.
+ */
+
+
+#include "gd.h"
+#include "gd_errors.h"
+#include "gdtest.h"
+#include <string.h>
+
+
+static void error_handler(int priority, const char *format, va_list args)
+{
+    char msg[256];

```

Comment:
AFAIK libpng makes no guarantee regarding the max length of the error messages, so, yes, we have to worry. Is `vsnprintf` portable enough for our purposes?